### ISSUE TITLE
Areas for React Coverage Improvement

### ISSUE THREAD

**Issue by jaredsilver** [empirical-org/Empirical-Core#2483]
Overall, we're starting to achieve pretty significant front-end coverage of our React components using Jest. However, there's still a long way to go before we get to where we really should be.

These are a few opportunities for improving coverage around critical application components. I recommend opening a new issue for each one individually when you decide to work on it and adding the issue number in parens next to the list item so others know it's claimed.

🔴 High Priority | 🔵 Medium Priority | ⚪️ Low Priority

- [ ] 🔴 Investigate writing tests for routers (`PublicUnitTemplatesRouter`, `LessonPlannerRouter`, `DiagnosticPlanner`, `DiagnosticReports`, etc.)
- [ ] 🔴 Improve container tests by adding coverage around ajax calls (blocked by #2482)
- [ ] 🔴 `client/app/bundles/HelloWorld/containers/__tests__/LessonPlanner.test.jsx` (partially blocked by #2482)
- [ ] 🔵 Test resource generators
- [ ] 🔵 Improve container tests by adding coverage around modules
- [ ] ⚪️ Improve test coverage by adding some massive mount tests that mount an entire container and work all the way down to the lowest rendered components
- [ ] ⚪️ Because speed was prioritized for writing initial tests, there are instances where arbitrary prop values were passed or required props were not passed (resulting in some angry messages from Jest). Run `npm run jest`, look for angry console error messages, and resolve them.
- [ ] ⚪️ Grep for `TODO` in all `.test.jsx` files
- [ ] ⚪️ Run `npm run jest:coverage` and get to work -- there's lots of room to improve! Specifically, look for components that have low coverage but are used in high traffic and/or mission critical areas of the site.
- [ ] ⚪️ Investigate speeding up Jest tests on Travis. They're taking _forever_ compared to local.
